smoke-tests/apps/LiveMetrics/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/LiveMetricsTest.java

```diff
@@ -27,14 +39,117 @@ abstract class LiveMetricsTest {
 
   @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
 
+  private boolean foundExceptionDoc = false;
+  private boolean foundTraceDoc = false;
+  private boolean foundDependency = false;
+  private boolean foundRequest = false;
+
   @Test
   @TargetUri("/test")
-  void doMostBasicTest() throws Exception {
-    testing.getTelemetry(0);
-
+  void testTelemetryDataFlow() throws java.lang.Exception {
     Awaitility.await()
-        .atMost(Duration.ofSeconds(10))
-        .until(() -> testing.mockedIngestion.isLiveMetricsPingReceived());
+        .atMost(Duration.ofSeconds(60))
+        .until(() -> testing.mockedIngestion.getCountForType("RequestData") == 1);
+
+    assertThat(testing.mockedIngestion.isPingReceived()).isTrue();
+
+    List<String> postBodies = testing.mockedIngestion.getPostBodies();
+    assertThat(postBodies).hasSizeGreaterThan(0); // should post at least once
+
+    for (String postBody : postBodies) {
+      searchPostBody(postBody);
+    }
+
+    assertThat(foundExceptionDoc).isTrue();
+    assertThat(foundTraceDoc).isTrue();
+    assertThat(foundDependency).isTrue();
+    assertThat(foundRequest).isTrue();
+  }
+
+  private void searchPostBody(String postBody) {
+    // Each post body is a list with a singular MonitoringDataPoint
+    List<MonitoringDataPoint> dataPoints = new ArrayList<>();
+    try {
+      JsonReader reader = JsonProviders.createReader(postBody);
+      dataPoints = reader.readArray(MonitoringDataPoint::fromJson);
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to parse post request body", e);
+    }
+
+    // Because the mock ping/posts should succeed, we should only have one MonitoringDataPoint per
+    // post
+    assertThat(dataPoints).hasSize(1);
+    MonitoringDataPoint dataPoint = dataPoints.get(0);
+
+    List<DocumentIngress> docs = dataPoint.getDocuments();
+    List<MetricPoint> metrics = dataPoint.getMetrics();
+
+    confirmDocsAreFiltered(docs);
+    confirmPerfCountersNonZero(metrics);
+    foundExceptionDoc = foundExceptionDoc || hasException(docs);
```

Comment:
I would avoid changing instance variables in the test methods.

If one day test methods are executed in parallel, the tests could become flaky.